src/lib/core/bidi/directionality.ts

```diff
@@ -0,0 +1,43 @@
+import {
+  EventEmitter,
+  Injectable,
+  Optional,
+  SkipSelf
+} from '@angular/core';
+
+export type Direction = 'ltr' | 'rtl';
+
+/**
+ * The directionality (LTR / RTL) context for the application (or a subtree of it).
+ * Exposes the current direction and a stream of direction changes.
+ */
+@Injectable()
+export class Directionality {
+  value: Direction = 'ltr';
+  public change = new EventEmitter<void>();
+
+  constructor() {
+    if (document) {
+    // TODO: handle 'auto' value -
+    // We still need to account for dir="auto".
+    // It looks like HTMLElemenet.dir is also "auto" when that's set to the attribute,
+    // but getComputedStyle return either "ltr" or "rtl". avoiding getComputedStyle for now
+    // though, we're already calling it for the theming check.
+    this.value =
+      (document.body.getAttribute('dir') ||
```

Comment:
Use the `dir` property instead of `getAttribute`? I believe they should be equivalent (but would be shorter)